assignments/minishell/README.md

```diff
@@ -0,0 +1,174 @@
+7
+```
+
+If the executable is located at the `/home/student/sum` absolute path, the following example should also be valid.
+
+```sh
+> /home/student/sum 2 4 1
+7
+```
+
+Each application will run in a separate child process of the minishell created using [fork](https://man7.org/linux/man-pages/man2/fork.2.html).
+
+### Environment Variables
+
+Your shell will support using environment variables.
+The environment variables will be initially inherited from the `bash` process that started your minishell application.
+
+If an undefined variable is used, its value is the empty string: `""`.
+
+> **_NOTE:_** The following examples contain comments which don't need to be supported by the minishell.
+> They are present here only to give a better understanding of the minishell's functionalities.
+
+```sh
+> NAME="John Doe"                    # Will assign "John Doe" to NAME variable
+> AGE=27                             # Will assign 27 to AGE variable
+> ./identify $NAME $LOCATION $AGE    # Will translate to ./identify "John Doe" "" 27 because $LOCATION is not defined
+```
+
+A variable can be assigned to another variable.
+
+```sh
+> OLD_NAME=$NAME    # Will assign the value of the NAME variable to OLD_NAME
+```
+
+### Operators
+
+#### Sequential Operator
+
+By using the `;` operator, you can chain multiple commands that will run sequentially, one after another.
+In the command `expr1; expr2` it is guaranteed that `expr1` will finish before `expr2` is be evaluated.
+
+```sh
+> echo "Hello"; echo "world!"; echo "Bye!"
+Hello
+world!
+Bye!
+```
+
+#### Parallel Operator
+
+By using the `&` operator you can chain multiple commands that will run in parallel.
+When running the command `expr1 & expr2`, both expressions are evaluated at the same time (by different processes).
+The order in which the two commands finish is not guaranteed.
+
+```sh
+> echo "Hello" & echo "world!" & echo "Bye!"  # The words may be printed in any order
+world!
+Bye!
+Hello
+```
+
+#### Pipe Operator
+
+With the `|` operator you can chain multiple commands so that the standard output of the first command is redirected to the standard input of the second command.
+
+Hint: Look into [anonymous pipes](https://man7.org/linux/man-pages/man2/pipe.2.html) and file descriptor inheritance while using [fork](https://man7.org/linux/man-pages/man2/fork.2.html).
+
+```sh
+> echo "Bye"                      # command outputs "Bye"
+Bye
+> ./reverse_input
+Hello                             # command reads input "Hello"
+olleH                             # outputs the reversed string "olleH"
+> echo "world" | ./reverse_input  # the output generated by the echo command will be used as input for the reverse_input executable
+dlrow
+```
+
+#### Chain Operators for Conditional Execution
+
+The `&&` operator allows chaining commands that are executed sequentially, from left to right.
+The chain of execution stops when it encounters the first command **that exists with an error (return code not 0)**.
```

Comment:
```suggestion
The chain of execution stops at the first command **that exists with an error (return code not 0)**.
```

I know it was me who made this suggestion a few days ago, but now I've found a better wording.